python/amici/sbml_import.py

```diff
@@ -363,7 +363,8 @@ def processSpecies(self):
         amounts = [spec.getInitialAmount() for spec in species]
 
         def getSpeciesInitial(index, conc):
-            if not math.isnan(conc):
+            if not (self.speciesHasOnlySubstanceUnits[index] or math.isnan(
```

Comment:
If speciesHasOnlySubstanceUnits is true, the initial amount should be set (and used), not the initial concentration. Is speciesHasOnlySubstanceUnits=True in combination with intialconcentration instead of initialamount valid SBML?

With change will use the nan, if the concentration is nan. I am not sure if this is what we want.

Reply:
In valid SBML, you can not use 
`speciesHasOnlySubstanceUnits = "True"` and set `intialConcentration`.
Already used nan, if the concentration was nan. And, moreover, if my understanding of the code is correct, the initial concentration will never be set to nan but rather to 0.0 ...
Which is the main source of our problems...

Reply:
Sorry, missed the parentheses after `not`.

Reply:
The main problem here is actually the libsbml-api, I think... the corresponding function getInitialConcentrations() is not doing the correct job... Yet, we have to handle it somehow...
This is the best quick-fix I could come up with that really helps in the situations we have, does not change the code for situations in which everything works and does not try to solve problems we currently don't have...

Reply:
> getInitialConcentrations() is not doing the correct job

What does it do and what should it do instead?

Reply:
To my understanding, it should return `nan`, if no initialConcentration is defined, but it returns `0.0`. That's misleading...

Reply:
Should first check `species.isSetInitialConcentration()` then.

Reply:
corrected it